server/gitlab/api.go

```diff
@@ -412,3 +412,35 @@ func (g *gitlab) ResolveNamespaceAndProject(
 	}
 	return "", "", ErrNotFound
 }
+
+func (g *gitlab) TriggerNewBuildPipeline(user *UserInfo, repo interface{}, refToCommit *string) (*internGitlab.Pipeline, error) {
+	var err error
+	client, err := g.gitlabConnect(*user.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		pipeline       *internGitlab.Pipeline
+		ctx, ctxCancel = context.WithTimeout(context.Background(), DefaultRequestTimeout)
+	)
+
+	defer ctxCancel()
+	errGroup, _ := errgroup.WithContext(ctx)
+	errGroup.Go(func() error {
+		p, response, gitError := client.Pipelines.CreatePipeline(repo, &internGitlab.CreatePipelineOptions{
+			Ref: refToCommit,
+		}, internGitlab.WithContext(ctx))
+
+		if gitError != nil && response != nil && response.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("failed to retrieve project by path: %w", gitError)
+		}
+		pipeline = p
+		return nil
+	})
+
+	if err = errGroup.Wait(); err != nil {
+		return nil, err
+	}
```

Comment:
I would love to see some kind of report from the Pipeline in Mattermost.

Reply:
the pipeline struct return in case of success contains the details related to build 

`type Pipeline struct {
	ID             int             `json:"id"`
	Status         string          `json:"status"`
	Ref            string          `json:"ref"`
	SHA            string          `json:"sha"`
	BeforeSHA      string          `json:"before_sha"`
	Tag            bool            `json:"tag"`
	YamlErrors     string          `json:"yaml_errors"`
	User           *BasicUser      `json:"user"`
	UpdatedAt      *time.Time      `json:"updated_at"`
	CreatedAt      *time.Time      `json:"created_at"`
	StartedAt      *time.Time      `json:"started_at"`
	FinishedAt     *time.Time      `json:"finished_at"`
	CommittedAt    *time.Time      `json:"committed_at"`
	Duration       int             `json:"duration"`
	Coverage       string          `json:"coverage"`
	WebURL         string          `json:"web_url"`
	DetailedStatus *DetailedStatus `json:"detailed_status"`
}`

Reply:
also inside the Detailed status there is an image as well

Reply:
Would you be interested in also adding a success message to this PR?

Reply:
makes sense.